### PR TITLE
Basic HTTP client functionality for raw data & JSON

### PR DESCRIPTION
**Pull request by hendrikebbers**
Basic HTTP functionality. This is needed for basic client session support and security

---
This change is [Reviewable](https://reviewable.io/reviews/canoo/dolphin-platform/590)

### CODE REVIEW

**Comment by hendrikebbers** [comment]
@timo0 this should be merged quite fast. Once this is done we can do the next release.

File: platform/dolphin-platform-client/src/main/java/com/canoo/dp/impl/platform/client/http/ByteArrayProvider.java [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package com.canoo.dp.impl.platform.client.http;
+
+/**
+ * Created by hendrikebbers on 19.09.17.
+ */
```
Comment (manish5196) [inline platform/dolphin-platform-client/src/main/java/com/canoo/dp/impl/platform/client/http/ByteArrayProvider.java]: Remove Comment and add Copy Right information.

File: platform/dolphin-platform-remoting-client/src/test/java/com/canoo/dolphin/client/impl/TestDolphinPlatformHttpClientConnector.java [hunk 2]
```diff
@@ -91,13 +100,15 @@ public String getHeaderField(String name) {
             }
         });
 
+        HttpClient httpClient = new HttpClientImpl(new Gson(), clientConfiguration.getConnectionFactory());
```
Comment (manish5196) [inline platform/dolphin-platform-remoting-client/src/test/java/com/canoo/dolphin/client/impl/TestDolphinPlatformHttpClientConnector.java]: Better to have a non-modifiable variable as final.

**Comment by hendrikebbers** [comment]
@kunsingh :see_no_evil:

File: platform/dolphin-platform-core/src/main/java/com/canoo/platform/core/DolphinRuntimeException.java [hunk 3]
```diff
@@ -27,24 +27,33 @@
...
     /**
      * Constructor
-     * @param thread Thread in that the exception happend
+     *
+     * @param thread  Thread in that the exception happend
      * @param message message of the exception
-     * @param cause cause of the exception
+     * @param cause   cause of the exception
      */
     public DolphinRuntimeException(Thread thread, String message, Throwable cause) {
```
Comment (kunsingh) [inline platform/dolphin-platform-core/src/main/java/com/canoo/platform/core/DolphinRuntimeException.java]: Make variables final.

File: platform/dolphin-platform-remoting-client/src/main/java/com/canoo/dp/impl/client/ClientContextImpl.java [hunk 4]
```diff
@@ -61,9 +70,12 @@
 
     private DolphinCommandHandler dolphinCommandHandler;
 
-    public ClientContextImpl(ClientConfiguration clientConfiguration, final Function<ClientModelStore, AbstractClientConnector> connectorProvider) {
```
Comment (kunsingh) [inline platform/dolphin-platform-remoting-client/src/main/java/com/canoo/dp/impl/client/ClientContextImpl.java]: Make variables final.

File: platform/dolphin-platform-remoting-server/src/main/java/com/canoo/dp/impl/server/context/CommunicationManager.java [hunk 5]
```diff
@@ -21,8 +21,8 @@
 public interface CommunicationManager {
 
     /**
-     * Returns true if commands should be send back to the client
-     * @return true if commands should be send back to the client
+     * Returns true if commands should be withContent back to the client
```
Comment (kunsingh) [inline platform/dolphin-platform-remoting-server/src/main/java/com/canoo/dp/impl/server/context/CommunicationManager.java]: Duplicate comment.

**Comment by hendrikebbers** [comment]
@manish5196 @kunsingh Are you done with the review?

File: platform/dolphin-platform-client/src/main/java/com/canoo/dp/impl/platform/client/http/ByteArrayProvider.java (hunk 1 above)
Comment (hendrikebbers) [reply platform/dolphin-platform-client/src/main/java/com/canoo/dp/impl/platform/client/http/ByteArrayProvider.java]: Done

File: platform/dolphin-platform-remoting-client/src/test/java/com/canoo/dolphin/client/impl/TestDolphinPlatformHttpClientConnector.java (hunk 2 above)
Comment (hendrikebbers) [reply platform/dolphin-platform-remoting-client/src/test/java/com/canoo/dolphin/client/impl/TestDolphinPlatformHttpClientConnector.java]: Done

File: platform/dolphin-platform-core/src/main/java/com/canoo/platform/core/DolphinRuntimeException.java (hunk 3 above)
Comment (hendrikebbers) [reply platform/dolphin-platform-core/src/main/java/com/canoo/platform/core/DolphinRuntimeException.java]: done

File: platform/dolphin-platform-remoting-client/src/main/java/com/canoo/dp/impl/client/ClientContextImpl.java (hunk 4 above)
Comment (hendrikebbers) [reply platform/dolphin-platform-remoting-client/src/main/java/com/canoo/dp/impl/client/ClientContextImpl.java]: done

File: platform/dolphin-platform-remoting-server/src/main/java/com/canoo/dp/impl/server/context/CommunicationManager.java (hunk 5 above)
Comment (hendrikebbers) [reply platform/dolphin-platform-remoting-server/src/main/java/com/canoo/dp/impl/server/context/CommunicationManager.java]: done

**Pull request merged**